### DISCUSSION TITLE
file_get_contents error for upload fields

### DISCUSSION THREAD

**Discussion by green17** [solspace/craft-freeform#1700] [Questions & Answers] [answered]
### What happened?

We recently upgraded from Freeform 3 to 4, along with upgrading Craft CMS from version 3 to 4. Our setup includes several file systems configured without public URLs for security reasons. When using the upload field with a standard, publicly accessible file system, everything functions as expected. However, when we use a file system without a public URL, the form fails to submit and generates an error - file_get_contents(test/file.pdf): Failed to open stream: No such file or directory.

Although the file is successfully uploaded to the correct location, the form will not submit.

### How can we reproduce this?

1. In Craft's filesystem settings, toggle the "Files in this filesystem have public URLs" to off for a filesystem.
2. In the form builder, choose this filesystem as the asset source for a file upload field.
3. When the form has loaded on the front end, choose a file to upload and submit the form.

### Freeform Edition

Pro

### Freeform Version

4.1.23

### Craft Version

Craft Pro 4.13.5

### When did this issue start?

After upgrading from older Freeform version

### Previous Freeform Version

3.x

**Comment by kjmartens** [comment]
Sorry for the trouble and the delay, @green17.

I tested this on my end and I can't seem to duplicate the issue. I'm curious, are you 100% sure the path is correct for the Asset/Filesystem? And how is your email notification template configured? 🙂

**Comment by green17** [comment]
Hi @kjmartens, the file system looks to be set up correctly as the file does upload in the background, however it does fail validation and will not submit (this happens when the field is set to required and when not). As this folder sits outside of the web folder, I have a path set up in DDEV (e.g. in the SNV file I have this variable: UPLOADS_PATH = /var/www/html/locked). 

When I tested it with a folder inside the web/assets directory, it also failed to submit though the file was uploaded to the correct location. The error was: deleteFile() is not implemented.

The email notification template is very basic, it really only loops through the submission fields. I have tried this without the notification and also without any other fields except for the file upload field.

**Comment by kjmartens** [comment]
Sorry for the delay @green17,

If you try to upload to the configured asset directory inside the Assets area in the Craft CP, does it work correctly? Or are you seeing any similar issues? Can you also attach a screenshot of your Freeform Diagnostics page? 🙂 

For good measure, can you screenshot what it looks like or attach the template file here? 🙂

**Comment by green17** [comment]
@kjmartens I have been able to upload assets via the CP, however when trying to upload via the form I now get a "deleteFile() is not implemented." error. Please see screenshots below:

![Screenshot 2025-01-13 at 14 17 19](https://github.com/user-attachments/assets/a8618e91-bd3c-4f0f-b1fc-9a4f02b97719)

![Screenshot 2025-01-13 at 14 18 12](https://github.com/user-attachments/assets/b8cfacd0-47f6-4f1c-a96f-51e98e289fa8)

In terms of the template, we aren't really doing anything aside from styling the forms. I have tested with the built-in templates (e.g. tailwind 3) and the error persisted.

**Reply by green17** [reply, marked as answer]
@kjmartens we managed to resolve this issue. Although it looked like an issue around how Craft was handling assets without public URLs, the underlying problem was a PDF transform plugin that we had installed. Once removed, everything worked as expected.

**Reply by kjmartens** [reply]
Thank you for confirming! 🙂
